### PR TITLE
OCPBUGS-19440: [release-4.14]: SriovNetwork, SriovIBNetwork, react on namespace creation

### PR DESCRIPTION
**Pull request by zeeke**
Backport of:
- https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/493

### CODE REVIEW

**Comment by zeeke** [comment]
@bn222 , @SchSeba Please take a look

File: controllers/sriovnetwork_controller.go [hunk 1]
```diff
@@ -173,8 +182,31 @@ func (r *SriovNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *SriovNetworkReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Reconcile when the target namespace is created after the SriovNetwork object.
```
Comment (bn222) [inline controllers/sriovnetwork_controller.go:185]: please also CC me next time on the upstream discussion.

Comment (zeeke) [reply controllers/sriovnetwork_controller.go:185]: Fine. Do you mean for every change in the operator? Or for controllers one? 
Most of the discussions are held in [upstream meetings](https://www.google.com/url?q=https://docs.google.com/document/d/1sJQMHbxZdeYJPgAWK1aSt6yzZ4K_8es7woVIrwinVwI/edit?usp%3Dsharing&sa=D&source=calendar&ust=1695915143499533&usg=AOvVaw21RJVl0ayZ4YWK60F4N4G7) on Monday and Tuesday

File: controllers/sriovnetwork_controller.go [hunk 2]
```diff
@@ -173,8 +182,31 @@ func (r *SriovNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Request
...
+					Info("Can't list SriovNetworks for namespace", "resource", e.Object.GetName(), "error", err)
+			}
+
+			for _, network := range networkList.Items {
+				q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+					Namespace: network.Namespace,
+					Name:      network.Name,
+				}})
+			}
+		},
```
[the remark this answers is not in the document]
Comment (zeeke) [reply controllers/sriovnetwork_controller.go:204]: I guess I can propose a refactor upstream

[the remark this answers is not in the document]
Comment (zeeke) [reply controllers/sriovnetwork_controller.go:204]: - https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/513

File: controllers/sriovnetwork_controller.go (hunk 1 above)
Comment (bn222) [reply controllers/sriovnetwork_controller.go:185]: Every change

**Comment by zeeke** [comment]
@SchSeba please take a look. We need backport-risk-assessed
@zhaozhanqi we also need cherry-pick-approved

**Comment by zhaozhanqi** [comment]
also cc @bn222 @wizhaoredhat for backport-risk-assessed
I will add cherry-pick-approved

**Pull request merged**